129: improved tests

davidabrahams opened the pull request:
closes #129 

removed setupTiles() and teardownTiles() function, now tests are not dependent on tiles.js functioning properly. If it isn't, only the tests in tiles.js will fail.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/chauncy-crib/tagprobot/199)
<!-- Reviewable:end -->

davidabrahams commented:
Review status: 0 of 8 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/helpers/map.spec.js, line 0 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAYk0AAUKXOM0ae-js:-KtAYhEbUEJW3ImTd8B1:b-522dci) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L0)):*
Note that this is actually a file you should review.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 5 of 8 files at r1.
Review status: 0 of 8 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[.babelrc, line 2 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfviscy28HrUdbSgT:-KtAfviscy28HrUdbSgU:bq18c2l) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/.babelrc#L2)):*
> ```JSON5
> {
>   "plugins": ["lodash", "babel-plugin-rewire"],
> ```

what does this do

---

*[package.json, line 37 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAd9RYOE64ZJOKoYOX:-KtAd9RZx12ybiRKucxc:b-wwtpnf) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/package.json#L37)):*
> ```JSON
>     "eslint-loader": "^1.9.0",
>     "eslint-plugin-import": "^2.7.0",
>     "sinon": "^2.4.1",
> ```

why did we change sinon and rewire to be regular deps?

---

*[src/tiles.js, line 77 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAg75g11EV9FvymDhv:-KtAg75g11EV9FvymDhw:bpycb1t) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/src/tiles.js#L77)):*
> ```JavaScript
>   if (!_.has(tileInfo, name)) {
>     return false;
>   }
> ```

Are we okay with this functionality of not asserting?

---

*[test/tiles.spec.js, line 149 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAe40oLGDuM0JXeVWZ:-KtAe40oLGDuM0JXeVW_:b3ym70c) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/tiles.spec.js#L149)):*
> ```JavaScript
>   setup();
> 
>   t.notOk(tileHasName(1, 'NAME_2'));
> ```

FYI I sorta think `t.false()` is more readable than `t.notOk()` (seeing as I just spent 5 seconds trying to figure that out) but now that I know that, it doesn't super matter to me

---

*[test/helpers/map.spec.js, line 11 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfhg7bsTuHk1axLdC:-KtAfhg7bsTuHk1axLdD:bdndr2y) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L11)):*
> ```JavaScript
>   __RewireAPI__ as MapRewireAPI,
> } from '../../src/helpers/map';
> import { teardownTiles } from '../tiles.spec';
> ```

Looks like you got a linter error for not using this

---

*[test/helpers/map.spec.js, line 125 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfTI6mVEELgNeZkhQ:-KtAfTI6mVEELgNeZkhR:blljf6l) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L125)):*
> ```JavaScript
>       'tileId',
>       ['ANGLE_WALL_1', 'ANGLE_WALL_2', 'ANGLE_WALL_3', 'ANGLE_WALL_4'],
>     ).returns(false);
> ```

Have you looked into stub().callsFake()? It might be better to make this a whole fake function instead of specifying what it should return with what specific args.

---

*[test/helpers/map.spec.js, line  at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAYk0AAUKXOM0ae-js:-KtAf8lw8b7ayGWv9O1Y:b-2rz5vq) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Note that this is actually a file you should review.
</blockquote></details>

FYI hahaha

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 0 of 10 files reviewed at latest revision, 7 unresolved discussions.

---

*[.babelrc, line 2 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfviscy28HrUdbSgT:-KtDVZhBz-l5CBkGyGck:bn863vp) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/.babelrc#L2)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

what does this do
</blockquote></details>

It's equivalent. `rewire` is an alias for `babel-plugin-rewire`. I figured it was more clear if we used the full name.

---

*[package.json, line 37 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAd9RYOE64ZJOKoYOX:-KtDViELuq-JOrxcgjVb:bdj27m7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/package.json#L37)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

why did we change sinon and rewire to be regular deps?
</blockquote></details>

I believe the opposite happened (I moved them to dev deps). They were regular deps before this PR, I realized that and fixed it in this PR.

---

*[src/tiles.js, line 77 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAg75g11EV9FvymDhv:-KtDVwqSNqScrE6MxrI1:b-bgip6u) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/src/tiles.js#L77)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Are we okay with this functionality of not asserting?
</blockquote></details>

Yeah when I reexamined it, I thought it make more sense for `tileHasName(0, 'POTATO')` to return false, rather than erroring. LMK if you disagree with this update.

---

*[test/tiles.spec.js, line 149 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAe40oLGDuM0JXeVWZ:-KtDWdgyRzbW3Lr8CIEX:bftisje) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/tiles.spec.js#L149)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

FYI I sorta think `t.false()` is more readable than `t.notOk()` (seeing as I just spent 5 seconds trying to figure that out) but now that I know that, it doesn't super matter to me
</blockquote></details>

I agree with you. For the sake of convention, I changed all `t.ok` and `t.notOk` to `t.true/false`

---

*[test/helpers/map.spec.js, line 11 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfhg7bsTuHk1axLdC:-KtDX1F78hkGmKsFpnri:bpuy5t4) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L11)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Looks like you got a linter error for not using this
</blockquote></details>

Yep. Thanks. Fixed

---

*[test/helpers/map.spec.js, line 125 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfTI6mVEELgNeZkhQ:-KtDXAi32wX26uwH4joX:br9168c) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L125)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Have you looked into stub().callsFake()? It might be better to make this a whole fake function instead of specifying what it should return with what specific args.
</blockquote></details>

It'll only get called with these specific args though. What would the `callsFake()` version look like? Also not that I look at it, I can just make it return false for all args in this test. Made that update.

---

*[test/helpers/map.spec.js, line  at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAYk0AAUKXOM0ae-js:-KtDWyGb67dAUudt6wm8:b2sb578) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

FYI hahaha
</blockquote></details>

Why is your disposition not Ok on this? :(

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 5 of 8 files at r1, 1 of 4 files at r2.
Review status: 0 of 10 files reviewed at latest revision, 9 unresolved discussions.

---

*[src/tiles.js, line 77 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAg75g11EV9FvymDhv:-KtGjfebyRNfHz7bHwkM:b-kxlmax) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/src/tiles.js#L77)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Yeah when I reexamined it, I thought it make more sense for `tileHasName(0, 'POTATO')` to return false, rather than erroring. LMK if you disagree with this update.
</blockquote></details>

I think I disagree. There shouldn't be a case when we pass in a name that doesn't exist and if we do, I might like to know. For example, if we pass in a tileId by accident.

---

*[test/tiles.spec.js, line 19 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtGlmwAyEd4tJ160M__:-KtGlmwAyEd4tJ160M_a:br6ilw9) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/tiles.spec.js#L19)):*
> ```JavaScript
> // functions for setup and teardown tests begin here
> 
> function setup() {
> ```

Do you want to specify this function name? Maybe `setupTiles()`?

---

*[test/tiles.spec.js, line 27 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtGlu9zJRrZ95ZLIt6d:-KtGlu9zJRrZ95ZLIt6e:b-bl3dzu) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/tiles.spec.js#L27)):*
> ```JavaScript
> }
> 
> function teardown() {
> ```

`teardownTiles()`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[package.json, line 37 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAd9RYOE64ZJOKoYOX:-KtIGdFyanvwIodLsDEw:b70hm6) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/package.json#L37)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

I believe the opposite happened (I moved them to dev deps). They were regular deps before this PR, I realized that and fixed it in this PR.
</blockquote></details>

OK i'm dumb :thumbsup: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 4 of 10 files reviewed at latest revision, 5 unresolved discussions.

---

*[test/helpers/map.spec.js, line 125 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfTI6mVEELgNeZkhQ:-KtIKTO7sSqAQtSqLhfA:bb6e0oh) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L125)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

It'll only get called with these specific args though. What would the `callsFake()` version look like? Also not that I look at it, I can just make it return false for all args in this test. Made that update.
</blockquote></details>

*now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[test/helpers/map.spec.js, line 125 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAfTI6mVEELgNeZkhQ:-KtIJqLMr0x4-ug7W06a:b-mls7rk) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L125)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

It'll only get called with these specific args though. What would the `callsFake()` version look like? Also not that I look at it, I can just make it return false for all args in this test. Made that update.
</blockquote></details>

My sleepy brain thought these three mocks were the same function, nevermind

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[test/helpers/map.spec.js, line  at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAYk0AAUKXOM0ae-js:-KtIHV8NZaEBGe3f8zQN:b-ix2luw) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Why is your disposition not Ok on this? :(
</blockquote></details>

FYI hmmm can you acknowledge this?  Piotr said "any acknowledgement will resolve an 'FYI' "

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 4 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[test/helpers/map.spec.js, line  at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAYk0AAUKXOM0ae-js:-KtIKnTbhM9E6fo4ugj2:b-z9l7rb) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/test/helpers/map.spec.js#L)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

FYI hmmm can you acknowledge this?  Piotr said "any acknowledgement will resolve an 'FYI' "
</blockquote></details>

didn't work lol

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 3 of 9 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[test/tiles.spec.js, line 19 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtGlmwAyEd4tJ160M__:-KtIU1GktwxXSptno5JL:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/tiles.spec.js#L19)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Do you want to specify this function name? Maybe `setupTiles()`?
</blockquote></details>

Done.

---

*[test/tiles.spec.js, line 27 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtGlu9zJRrZ95ZLIt6d:-KtIU1lJGA9DChRSb-ro:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/tiles.spec.js#L27)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`teardownTiles()`?
</blockquote></details>

Done.

---

*[src/tiles.js, line 77 at r1](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtAg75g11EV9FvymDhv:-KtISyrmINYG08WcI57o:b-qssebl) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5b3726ec650496010b147f09f1c12757db6e51ef/src/tiles.js#L77)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

I think I disagree. There shouldn't be a case when we pass in a name that doesn't exist and if we do, I might like to know. For example, if we pass in a tileId by accident.
</blockquote></details>

fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 1 of 8 files at r1, 3 of 4 files at r2.
Review status: 3 of 9 files reviewed at latest revision, 29 unresolved discussions, some commit checks failed.

---

*[test/helpers/map.spec.js, line 101 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHbUs5nbEiqIqhqpbk:-KtHbUs5nbEiqIqhqpbl:b-indkmf) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L101)):*
> ```JavaScript
> test('getTileTraversabilityInCells() ', tester => {
>   tester.test('returns correctly with entirely traversable tile, CPTL=1', t => {
>     const mockGetTileProperty = sinon.stub().withArgs('tileId', 'traversable').returns(true);
> ```

comment `// is fully traversable`

---

*[test/helpers/map.spec.js, line 105 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHYBJlIxaDqw-U5SBp:-KtHYBJlIxaDqw-U5SBq:be1s0hf) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L105)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('getTileProperty', mockGetTileProperty);
> 
>     t.same(getTileTraversabilityInCells('tileId'), [
> ```

Change `'tileId'` to `'mockTileId'`?

---

*[test/helpers/map.spec.js, line 116 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHl0pg8Sl33LkX0VK2:-KtHl0pg8Sl33LkX0VK3:bgx8fop) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L116)):*
> ```JavaScript
> 
>   tester.test('returns correctly with entirely nontraversable tile, CPTL=1', t => {
>     // not fully traversable
> ```

FYI
Super nit: I like `entirely traversable` better than `fully traversable`. For some reasons I have to think for a second when I read `fully traversable`. You can leave it if you want, but if you change it please change it in all cases.

---

*[test/helpers/map.spec.js, line 139 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHdZu1vDIEx0sST06m:-KtHdZu1vDIEx0sST06n:bibovml) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L139)):*
> ```JavaScript
> 
> 
>   tester.test('returns correctly with entirely traversable tile, CPTL=4', t => {
> ```

`'returns correctly with entirely nontraversable tile, CPTL=4'`

---

*[test/helpers/map.spec.js, line 151 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHdhcDFoLbijYGkthu:-KtHdhcDFoLbijYGkthv:b-j45v16) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L151)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
> 
>     t.same(getTileTraversabilityInCells('tileId'), [
> ```

`'tileId'` to `'mockTileId'`

---

*[test/helpers/map.spec.js, line 166 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHo6hZR1GaSGav5mIZ:-KtHo6hZR1GaSGav5mI_:b-eqeae) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L166)):*
> ```JavaScript
> 
> 
>   tester.test('returns correctly with CNTO and small radius, CPTL=4', t => {
> ```

`'returns correctly with CNTO, radius=14, CPTL=4'`?

---

*[test/helpers/map.spec.js, line 198 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHeiErwtpbpoPMx1SS:-KtHeiErwtpbpoPMx1ST:b1q4akp) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L198)):*
> ```JavaScript
> 
> 
>   tester.test('returns correctly with CNTO and false radius, CPTL=4', t => {
> ```

`'returns correctly with CNTO, radius=15, CPTL=4'`?

---

*[test/helpers/map.spec.js, line 212 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHf45UZeL8cfu3Mx8L:-KtHf45UZeL8cfu3Mx8M:b-qb3qz7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L212)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('getTileProperty', mockGetTileProperty);
>     MapRewireAPI.__Rewire__('tileHasProperty', mockTileHasProperty);
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
> ```

Is the test whitespace convention now setup, newline, test, newline, teardown? If so, add a newline here.

---

*[test/helpers/map.spec.js, line 232 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHkWeoYQ14PIJlxwnY:-KtHkWeoYQ14PIJlxwnZ:btbd96o) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L232)):*
> ```JavaScript
>     // not fully traversable
>     const mockGetTileProperty = sinon.stub().withArgs('tileId', 'traversable').returns(false);
>     // does not has radius
> ```

"does not has radius"...

-_-

---

*[test/helpers/map.spec.js, line 246 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHkS2JTxliRrh8BZBZ:-KtHkS2JTxliRrh8BZB_:b-4yr6dt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L246)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('tileHasProperty', mockTileHasProperty);
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
>     MapRewireAPI.__Rewire__('tileHasName', mockTileHasName);
> ```

newline

---

*[test/helpers/map.spec.js, line 266 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHks0rANfIOv0rLRj0:-KtHks0rANfIOv0rLRj1:b-wasw3w) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L266)):*
> ```JavaScript
>     // not fully traversable
>     const mockGetTileProperty = sinon.stub().withArgs('tileId', 'traversable').returns(false);
>     // does not has radius
> ```

`// hasn't not got a radius`

---

*[test/helpers/map.spec.js, line 280 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnOvG0dxfhBdS63TJ:-KtHnOvG0dxfhBdS63TK:b-4yr6dt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L280)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('tileHasProperty', mockTileHasProperty);
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
>     MapRewireAPI.__Rewire__('tileHasName', mockTileHasName);
> ```

newline

---

*[test/helpers/map.spec.js, line 300 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHmbt4rAa3tSTTj3Pf:-KtHmbt4rAa3tSTTj3Pg:b-qff9tz) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L300)):*
> ```JavaScript
>     // not fully traversable
>     const mockGetTileProperty = sinon.stub().withArgs('tileId', 'traversable').returns(false);
>     // does not has radius
> ```

`// radius is what it has not`

---

*[test/helpers/map.spec.js, line 314 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnNtDvP8uVk8NIknH:-KtHnNtDvP8uVk8NIknI:b-4yr6dt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L314)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('tileHasProperty', mockTileHasProperty);
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
>     MapRewireAPI.__Rewire__('tileHasName', mockTileHasName);
> ```

newline

---

*[test/helpers/map.spec.js, line 334 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHmg-RWKuJAQhHFstM:-KtHmg-RWKuJAQhHFstN:buip14q) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L334)):*
> ```JavaScript
>     // not fully traversable
>     const mockGetTileProperty = sinon.stub().withArgs('tileId', 'traversable').returns(false);
>     // does not has radius
> ```

`// does radius, but not has`

---

*[test/helpers/map.spec.js, line 348 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnMdSN4f7fLopie01:-KtHnMdThqIGlNy52Eod:b-4yr6dt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L348)):*
> ```JavaScript
>     MapRewireAPI.__Rewire__('tileHasProperty', mockTileHasProperty);
>     MapRewireAPI.__Rewire__('tileIsOneOf', mockTileIsOneOf);
>     MapRewireAPI.__Rewire__('tileHasName', mockTileHasName);
> ```

newline

---

*[test/helpers/map.spec.js, line 365 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHneAxSjS4ZVDM-uL8:-KtHneAxSjS4ZVDM-uL9:b-cwdo3s) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L365)):*
> ```JavaScript
> 
> 
>   tester.test('returns correctly with nontraversable tile, CPTL=8', t => {
> ```

`'returns correctly with CNTO, radius=8, CPTL=8'`?

Should this test be moved up above the angle wall tests?

This should likely be broken up into two tests, as per the example above where we tested two different radii.

---

*[test/helpers/map.spec.js, line 423 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHpKJ1X3NDNh90Ov_1:-KtHpKJ1X3NDNh90Ov_2:b-wsyb30) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L423)):*
> ```JavaScript
> 
> 
> test('getMapTraversabilityInCells: returns correctly with CPTL=1', t => {
> ```

Shouldn't these tests be nested?

---

*[test/helpers/map.spec.js, line 423 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIoZyIchbAEt8oA2gt:-KtIoZyIchbAEt8oA2gu:b-gz81le) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L423)):*
> ```JavaScript
> 
> 
> test('getMapTraversabilityInCells: returns correctly with CPTL=1', t => {
> ```

Maybe add some detail to this test's description? It's pretty big and does more than just check return values.

---

*[test/helpers/map.spec.js, line 509 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIt_t54SiOpE49VarJ:-KtIt_t54SiOpE49VarK:b-9cq1k4) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L509)):*
> ```JavaScript
> 
> 
> test('getMapTraversabilityInCells: returns correctly with CPTL=3', t => {
> ```

FYI

you could also add more description to this test (or leave it if you like)

---

*[test/helpers/map.spec.js, line 510 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIsO6LJX3AROslehbM:-KtIsO6LJX3AROslehbN:b-tf5rnt) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L510)):*
> ```JavaScript
> 
> test('getMapTraversabilityInCells: returns correctly with CPTL=3', t => {
>   MapRewireAPI.__Rewire__('CPTL', 3);
> ```

Could you change this to a `CPTL` that can exist? 4?

`40%3 !== 0`

---

*[test/helpers/map.spec.js, line 511 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIsGiqbSRLkNqMq_Xw:-KtIsGiqbSRLkNqMq_Xx:b3kre) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L511)):*
> ```JavaScript
> test('getMapTraversabilityInCells: returns correctly with CPTL=3', t => {
>   MapRewireAPI.__Rewire__('CPTL', 3);
>   // create a dummy map. Assume all objects 
> ```

?

---

*[test/helpers/map.spec.js, line 558 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtItqt4vfTRXUEcg8hs:-KtItqt4vfTRXUEcg8ht:b-w1392b) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L558)):*
> ```JavaScript
>   MapRewireAPI.__ResetDependency__('updateNTSprites');
>   MapRewireAPI.__ResetDependency__('getTileTraversabilityInCells');
> ```

Remove newline

---

*[test/helpers/map.spec.js, line 620 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIuvsYLl9YvakW3olp:-KtIuvsYLl9YvakW3olq:byszoqj) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L620)):*
> ```JavaScript
>     ]);
>     t.same(mockTilesToUpdateValues, [tempNT, tempT, tempT, tempNT]);
>     t.true(mockGeneratePermanentNTSprites.calledThrice); // three permNT objects
> ```

can you make this be `t.true(mockGeneratePermanentNTSprites.callCount, 3);`

If you used things like calledOnce or anything anywhere, I think I would prefer them to be `callCount, N.

---

*[test/helpers/map.spec.js, line 635 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIvKEkTrQ4LnfAMyag:-KtIvKEkTrQ4LnfAMyah:bxsshsd) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L635)):*
> ```JavaScript
>   });
>   tester.end();
> });
> ```

FYI

Holy. Tests. Batman.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 1 of 2 files at r3.
Review status: 3 of 9 files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 1 of 2 files at r3.
Review status: 3 of 9 files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 3 of 9 files reviewed at latest revision, 27 unresolved discussions, some commit checks failed.

---

*[test/helpers/map.spec.js, line 101 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHbUs5nbEiqIqhqpbk:-KtKZytkGp-fZg6wgUpb:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L101)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

comment `// is fully traversable`
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 105 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHYBJlIxaDqw-U5SBp:-KtKZzN1Q9MXDwJJNUj9:b-ik3ckv) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L105)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Change `'tileId'` to `'mockTileId'`?
</blockquote></details>

Done. Changed everywhere

---

*[test/helpers/map.spec.js, line 116 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHl0pg8Sl33LkX0VK2:-KtK_OpbQaUpbA0Sf3oU:bzd19hl) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L116)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI
Super nit: I like `entirely traversable` better than `fully traversable`. For some reasons I have to think for a second when I read `fully traversable`. You can leave it if you want, but if you change it please change it in all cases.
</blockquote></details>

Done. everywhere.

---

*[test/helpers/map.spec.js, line 139 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHdZu1vDIEx0sST06m:-KtK_YBUJQDeEasvzc2l:bq5d6ui) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L139)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`'returns correctly with entirely nontraversable tile, CPTL=4'`
</blockquote></details>

thanks good catch

---

*[test/helpers/map.spec.js, line 151 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHdhcDFoLbijYGkthu:-KtK__YO3iass4jGcaif:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L151)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`'tileId'` to `'mockTileId'`
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 166 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHo6hZR1GaSGav5mIZ:-KtK_oArajnnf1errBiB:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L166)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`'returns correctly with CNTO, radius=14, CPTL=4'`?
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 198 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHeiErwtpbpoPMx1SS:-KtK_oZZn3ijnNFpEHe5:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L198)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`'returns correctly with CNTO, radius=15, CPTL=4'`?
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 212 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHf45UZeL8cfu3Mx8L:-KtK_tE2oSdACEaDV3lu:b-q9k7f7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L212)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Is the test whitespace convention now setup, newline, test, newline, teardown? If so, add a newline here.
</blockquote></details>

Done. That's what I was aiming for. Your a champion on these code reviews

---

*[test/helpers/map.spec.js, line 232 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHkWeoYQ14PIJlxwnY:-KtK_wrXa3Rt4AN3M61m:b-qc8gqc) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L232)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

"does not has radius"...

-_-
</blockquote></details>

sorry.... fixed

---

*[test/helpers/map.spec.js, line 246 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHkS2JTxliRrh8BZBZ:-KtKhhBCytEQ94Aj2FtD:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L246)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

newline
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 266 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHks0rANfIOv0rLRj0:-KtKa59IlD2UNXvzt_SI:blq7duh) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L266)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`// hasn't not got a radius`
</blockquote></details>

HAHA. alright great.

---

*[test/helpers/map.spec.js, line 280 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnOvG0dxfhBdS63TJ:-KtKaAu1YixGl4bRntxO:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L280)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

newline
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 300 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHmbt4rAa3tSTTj3Pf:-KtKaBvdklCSzP-0fGHj:b2nhipw) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L300)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`// radius is what it has not`
</blockquote></details>

LOL

---

*[test/helpers/map.spec.js, line 314 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnNtDvP8uVk8NIknH:-KtKhmoxTau8-Ch5rpKW:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L314)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

newline
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 334 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHmg-RWKuJAQhHFstM:-KtKhrkA8lH9e06jcqsc:bdj75cv) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L334)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`// does radius, but not has`
</blockquote></details>

:frog: 

---

*[test/helpers/map.spec.js, line 348 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHnMdSN4f7fLopie01:-KtKhvL8HpH4B9FaT46R:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L348)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

newline
</blockquote></details>

Done.

---

*[test/helpers/map.spec.js, line 365 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHneAxSjS4ZVDM-uL8:-KtKaQCZOrNjaI3gFUuv:b-8bnam0) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L365)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

`'returns correctly with CNTO, radius=8, CPTL=8'`?

Should this test be moved up above the angle wall tests?

This should likely be broken up into two tests, as per the example above where we tested two different radii.
</blockquote></details>

Done. Thanks for typing those out. I've just been copy pasting them in

---

*[test/helpers/map.spec.js, line 423 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtHpKJ1X3NDNh90Ov_1:-KtKcA_h3K9yXKAzBzxF:b-hqc1qu) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L423)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Shouldn't these tests be nested?
</blockquote></details>

They are now nested! Have fun with a huge diff cuz all the lines are indented with two extra spaces now ;)

---

*[test/helpers/map.spec.js, line 423 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIoZyIchbAEt8oA2gt:-KtKdYjeUZYoUMmVWjaB:b-euxdke) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L423)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Maybe add some detail to this test's description? It's pretty big and does more than just check return values.
</blockquote></details>

Good call. Sad thing is, test description is now quite long :(

---

*[test/helpers/map.spec.js, line 509 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIt_t54SiOpE49VarJ:-KtKdi8YCcsWXPGBuAnQ:bqw14vv) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L509)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI

you could also add more description to this test (or leave it if you like)
</blockquote></details>

Copied description from first one

---

*[test/helpers/map.spec.js, line 510 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIsO6LJX3AROslehbM:-KtKdrecZlfvYhnH7MFn:bn4gsh7) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L510)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Could you change this to a `CPTL` that can exist? 4?

`40%3 !== 0`
</blockquote></details>

yep. fixed.

---

*[test/helpers/map.spec.js, line 511 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIsGiqbSRLkNqMq_Xw:-KtKeXyOKa5FQ2zzM4lC:bh7nat1) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L511)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

?
</blockquote></details>

Yeah I just saw that. I think i fixed it to what I meant.

---

*[test/helpers/map.spec.js, line 558 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtItqt4vfTRXUEcg8hs:-KtKeohSwH6NWlgoDmAT:bqyiiei) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L558)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Remove newline
</blockquote></details>

:dango: 

---

*[test/helpers/map.spec.js, line 620 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIuvsYLl9YvakW3olp:-KtKfIPaDR5srL3UsNR9:b-9ghatq) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L620)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

can you make this be `t.true(mockGeneratePermanentNTSprites.callCount, 3);`

If you used things like calledOnce or anything anywhere, I think I would prefer them to be `callCount, N.
</blockquote></details>

Only place it shows up in this file. But it shows up everywhere everywhere else. And I fixed them all.

---

*[test/helpers/map.spec.js, line 635 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIvKEkTrQ4LnfAMyag:-KtKhyXG0cmpeuFRDMWH:b-us6bha) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L635)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI

Holy. Tests. Batman.
</blockquote></details>

:potato: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtN9uUMXO9EibX5nKod:-KtN9uUMXO9EibX5nKoe:b7jvtai) (no related file):*
> 129: changing all t.ok to t.true, and t.notOk to t.false

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 3 of 3 files at r4.
Review status: 3 of 9 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[test/tiles.spec.js, line 56 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtNCaO7dQl8b6v0fZOB:-KtNCaO7dQl8b6v0fZOC:b4tp6gs) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/4069ae2b0712b4bea2413bf0fc8810a951c5d996/test/tiles.spec.js#L56)):*
> ```JavaScript
>   t.is(mockTileInfo.BLUE_GATE.traversable, true);
>   // tileInfo dependent on team color
>   t.is(mockAmRed.callCount, 1);
> ```

@shane why did you want this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[test/tiles.spec.js, line 56 at r4](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtNCaO7dQl8b6v0fZOB:-KtNE3bphbD8uW8lQ01r:bq438d6) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/4069ae2b0712b4bea2413bf0fc8810a951c5d996/test/tiles.spec.js#L56)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

@shane why did you want this
</blockquote></details>

OK resolving this to put this conversation in one thread

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 1 of 8 files at r1, 3 of 4 files at r2, 1 of 2 files at r3, 3 of 3 files at r4.
Review status: 3 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[test/helpers/map.spec.js, line 620 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIuvsYLl9YvakW3olp:-KtNDpaPm-YqYk1MLd8h:b-wmc9tq) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L620)):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Only place it shows up in this file. But it shows up everywhere everywhere else. And I fixed them all.
</blockquote></details>

Can I get some reasoning on this @shanek21 before I accept it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 8 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtN9uUMXO9EibX5nKod:-KtNEefOdRV2eG9fYgmF:b-oq79rw) (no related file):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

> 129: changing all t.ok to t.true, and t.notOk to t.false
</blockquote></details>

fixed thats for catching that

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[test/helpers/map.spec.js, line 620 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIuvsYLl9YvakW3olp:-KtNHmFLSZjfF8w6u-0N:b-rvotd6) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L620)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Can I get some reasoning on this @shanek21 before I accept it?
</blockquote></details>

Convention

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[test/helpers/map.spec.js, line 620 at r2](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtIuvsYLl9YvakW3olp:-KtNSY-lvdqMLkZmPaR8:b-4i3qau) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/d22040dcd19c05c8fd51e7298cd67db6a4f437c3/test/helpers/map.spec.js#L620)):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

Convention
</blockquote></details>

OK sounds good

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/helpers/map.js, line 94 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtNSr11xrtEA2Bbwdr7:-KtNSr11xrtEA2Bbwdr8:bfg2q88) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5bdc25f19e0761f0b6e01d1c2e953b061a5369d3/src/helpers/map.js#L94)):*
> ```JavaScript
>         assert(
>           tileIsOneOf(tileId, ['ANGLE_WALL_1', 'ANGLE_WALL_2', 'ANGLE_WALL_3', 'ANGLE_WALL_4']),
>           'tile expected to be angle wall');
> ```

Put `); on newline https://github.com/airbnb/javascript#functions--signature-invocation-indentation

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 1 of 8 files at r1.
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/helpers/map.js, line 94 at r5](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtNSr11xrtEA2Bbwdr7:-KtP6SY2yrVyg5AJ7sCr:b-896fix) ([raw file](https://github.com/chauncy-crib/tagprobot/blob/5bdc25f19e0761f0b6e01d1c2e953b061a5369d3/src/helpers/map.js#L94)):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Put `); on newline https://github.com/airbnb/javascript#functions--signature-invocation-indentation
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Reviewed 1 of 1 files at r6.
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPJKKGdcQYvOHcfNFq:-KtPJKKGdcQYvOHcfNFr:byfsf8q) (no related file):*
FYI

For this commit message:
`129: nest tests. provide better descriptions for tests. Fix typos.
Remove calledOnce tests in favor of callCount`
Do you think it would be better to list things in a comma separated list that way we could avoid entire sentence structures (punctuation, capitalization, etc)?
Ex, I like this one:
`129: move dependencies to dev, update package-lock, and use full name
for babel-plugin-rewire in .babelrc`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
Reviewed 1 of 1 files at r6.
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPJKKGdcQYvOHcfNFq:-KtPKiGzAgXlG_mXCXEG:bpeqr) (no related file):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

FYI

For this commit message:
`129: nest tests. provide better descriptions for tests. Fix typos.
Remove calledOnce tests in favor of callCount`
Do you think it would be better to list things in a comma separated list that way we could avoid entire sentence structures (punctuation, capitalization, etc)?
Ex, I like this one:
`129: move dependencies to dev, update package-lock, and use full name
for babel-plugin-rewire in .babelrc`
</blockquote></details>

Yeah I agree.  I also don't think the last "and" is necessary either but judgement call.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPLS7LE22il_bmOIsR:-KtPLS7LE22il_bmOIsS:bvqyow7) (no related file):*
If we're being particular about commits, would it make sense to squash "refactor setup->setupTiles, teardown->teardownTiles" into a previous commit?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

billmwong commented:
*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPLS7LE22il_bmOIsR:-KtPPsEmxuxsQgJudAra:b-i317aa) (no related file):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

If we're being particular about commits, would it make sense to squash "refactor setup->setupTiles, teardown->teardownTiles" into a previous commit?
</blockquote></details>

LGTM david gave me a valid reason for not squashing <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPJKKGdcQYvOHcfNFq:-KtPR_oSl4AZ78RSsugl:b9lto2x) (no related file):*
<details><summary><i>Previously, billmwong (Bill Wong) wrote…</i></summary><blockquote>

Yeah I agree.  I also don't think the last "and" is necessary either but judgement call.
</blockquote></details>

Done on both

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

shanek21 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPJKKGdcQYvOHcfNFq:-KtRoQ2Dqu6a_-rD7Vj5:b-hn9a2o) (no related file):*
<details><summary><i>Previously, davidabrahams wrote…</i></summary><blockquote>

Done on both
</blockquote></details>

I think billo wants you to remove the `and` from `129: move dependencies to dev, update package-lock, and use full name for babel-plugin-rewire in .babelrc`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->

davidabrahams commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[a discussion](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199#-KtPJKKGdcQYvOHcfNFq:-KtRpqyUh0WkJ_YQl9-G:b-87u3st) (no related file):*
<details><summary><i>Previously, shanek21 (Shane Kelly) wrote…</i></summary><blockquote>

I think billo wants you to remove the `and` from `129: move dependencies to dev, update package-lock, and use full name for babel-plugin-rewire in .babelrc`
</blockquote></details>

I did

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/chauncy-crib/tagprobot/199)*
<!-- Sent from Reviewable.io -->
